pulp_puppet_plugins/pulp_puppet/forge/releases.py

```diff
@@ -50,24 +83,66 @@ def view(consumer_id, repo_id, module_name, version=None):
         repo_ids = get_bound_repos(consumer_id)
     else:
         repo_ids = [repo_id]
-    if version:
-        unit = find_version(repo_ids, module_name, version)
-    else:
-        unit = find_newest(repo_ids, module_name)
-    if not unit:
-        raise web.NotFound()
+
+    dbs = None
+    return_data = None
     try:
-        data = unit.build_dep_metadata()
+        # Get the list of database files to query
+        dbs = get_repo_data(repo_ids)
+
+        # Build list of units to return
+        ret = []
+        # If a version was specified filter by that specific version of the module
+        if version:
+            for unit in unit_generator(dbs, module_name):
+                if unit.version == version:
+                    ret.append(unit)
+                    break
+        else:
+            units = [individual_unit for individual_unit in unit_generator(dbs, module_name)]
```

Comment:
You can skip the list comprehension and just cast it.

``units = list(unit_generator(dbs, module_name))``